src/keys/bip39.rs

```diff
@@ -43,17 +151,39 @@ pub mod wordlist {
     #[cfg_attr(docsrs, doc(cfg(feature = "bip39-jp")))]
     include!("bip39.jp.rs");
 
-    #[derive(Debug, PartialEq, Eq)]
-    pub enum Error {
-        InvalidEntropyCount(usize),
-        NoSuchWord(String),
-        ChecksumMismatch,
-    }
-
     const fn cs(ent: usize) -> usize {
         ent / 32
     }
 
+    impl<'a> Wordlist<'a> {
+        // TODO: should it be pub?
+        const fn new_unchecked(separator: char, words: &'a [&'a str; 2048]) -> Self {
+            Self { words, separator }
+        }
+
+        pub fn new(separator: char, words: &'a [&'a str; 2048]) -> Result<Self, Error> {
+            // normalize separator char
+            let mut s = String::new();
+            s.push(separator);
+            let s: String = s.nfkd().collect();
+
+            if s.chars().count() == 1 {
+                let separator = s.chars().next().unwrap();
```

Comment:
Do an `if let Some(char) = s.chars().next()` instead if .count() == 1 then unwrap

Reply:
This is a workaround to normalize a single char. unicode_normalization crate doesn't provide such function. Here we want to match "exactly one normalized char", not "at least one char". Rewritten